src/rofiles-fuse/main.c

```diff
@@ -286,6 +333,9 @@ callback_truncate (const char *path, off_t size)
 static int
 callback_utimens (const char *path, const struct timespec tv[2])
 {
+  /* This one isn't write-verified, we support changing times
+   * even for hardlinked files.
```

Comment:
Can you expand on this? Don't we need to keep bare objects at `OSTREE_TIMESTAMP` to be consistent?

Reply:
It's an interesting point...gets a bit into a philosophical question of whether rofiles-fuse is for "anything using hardlinks" or whether it's really specific to ostree.  The "anything using hardlinks" case might be a backup system or non-os-data object store was my thought originally.

For example I could probably have "object semantics" for my `~/Documents` which has a bunch of random PDFs and `.txt` and LibreOffice files etc. that should really never be overwritten in place.  OTOH I've been thinking about storing all of that in git.

So...hmm.  It's not a new issue here and we should think about changing this, but I worry a bit about the backcompat issues.  In practice nothing we use rofiles-fuse for should be touching timestamps for files it didn't create at the time I would guess.

Reply:
> For example I could probably have "object semantics" for my ~/Documents which has a bunch of random PDFs and .txt and LibreOffice files etc. that should really never be overwritten in place. OTOH I've been thinking about storing all of that in git.

I've been using [git-annex](https://git-annex.branchable.com/) for this for some time now. Good news, it's in fedora.

Reply:
Right, git-annex is likely an excellent example of a possible user of rofiles-fuse semantics, specifically [its direct mode](https://git-annex.branchable.com/direct_mode/).  Though I didn't look closely.